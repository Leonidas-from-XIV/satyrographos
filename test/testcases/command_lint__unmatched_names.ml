open Core
open Satyrographos_testlib

let opam_file_with_name_field =
  "satysfi-package.opam",
  TestLib.opam_file_for_test
    ~name:"satysfi-package-mark-ii"
    ~version:"0.1"
    ()

let opam_file_without_name_field =
  "satysfi-package-document.opam", TestLib.opam_file_for_test
    ?name:None
    ~version:"0.1"
    ~depends:{|
  "satysfi" {>= "0.0.5" & < "0.0.6"}
  "satyrographos" {>= "0.0.2.6" & < "0.0.3"}

  "satysfi-package" {= "0.1"}
|}
    ()

let satyristes =
  "Satyristes", sprintf
    {|(version "0.0.2")

(library
  (name "package")
  (version "0.1")
  (sources ())
  (opam "satysfi-package.opam")
  (dependencies ()))

(libraryDoc
  (name "package-doc")
  (version "0.1")
  (build ())
  (sources ())
  (opam "satysfi-package-document.opam")
  (dependencies ((package ()))))
|}

let files =
  [ opam_file_with_name_field;
    opam_file_without_name_field;
    satyristes;
  ]

let opam_libs = Satyrographos.Library.[
    {empty with name = Some "package"};
  ]

let () =
  TestCommand.test_lint_command ~opam_libs files
