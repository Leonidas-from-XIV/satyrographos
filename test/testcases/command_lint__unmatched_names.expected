@@test_library@@/Satyristes:3:1: (module package):
@@test_library@@/satysfi-package.opam:
Error: opam-file/name
  OPAM package name “satysfi-package-mark-ii” should be “satysfi-package”.

@@test_library@@/Satyristes:10:1: (module package-doc):
@@test_library@@/satysfi-package-document.opam:
Error: opam-file/name
  OPAM package name “satysfi-package-document” should be “satysfi-package-doc”.

2 problem(s) found.
